### PR TITLE
17158 update fetchNameRequest

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#17158

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-540-bi05xbsj.web.app

File: src/mixins/name-request-mixin.ts
Comment (severinbeauvais) [inline src/mixins/name-request-mixin.ts:29]: Should this handle the different kinds of errors (missing phone/email, invalid phone/email, etc)?

What does the UI show if I search for a NR without phone/email, or with invalid phone/email?

Comment (severinbeauvais) [reply src/mixins/name-request-mixin.ts:29]: Oh, wait, I see you did that in LegalServices! Nice.

But doesn't the error handling above override the error from LegalServices?

**Comment by severinbeauvais** [comment]
Kevin, please add some other reviewers (eg, James, Shaoyun). I will wait for them to review before I approve. Thanks.

**Review by JazzarKarim — APPROVED** [review]
LGTM Kevin! 👍  Please wait for another approval or two before merging.

File: src/mixins/name-request-mixin.ts
Comment (kzdev420) [reply src/mixins/name-request-mixin.ts:29]: Yes, I agree
I think we can remove the above error handling.
What do you think?

Comment (severinbeauvais) [reply src/mixins/name-request-mixin.ts:29]: I'm not sure. Why don't you simulate some errors and see what happens?

I think you need to keep the "emit" so that a dialog is displayed.

Comment (kzdev420) [reply src/mixins/name-request-mixin.ts:29]: Display like this when we get the errors from API.

Screenshot 2023-11-09 at 7 51 05 AM

Comment (severinbeauvais) [reply src/mixins/name-request-mixin.ts:29]: The error handling above only handles a single case (not_found).

The error handling you deleted below previously handled incorrect_email and incorrect_phone.

However, the dialog previously displayed the same thing for all 3 cases. Furthermore, you no longer use the incorrect_email or incorrect_phone.

I think you should retain the error case from the API call up to the dialog. To do that, I would move the response handling from fetchNameRequest() to here, and I'd replace incorrect_email and incorrect_phone with new enums. This design would allow the dialog to decide what message to display (and I wouldn't change that) instead of losing the different cases in the underlying logic. Does this make sense?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-540-bi05xbsj.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-540-bi05xbsj.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-540-bi05xbsj.web.app

File: src/mixins/name-request-mixin.ts
Comment (kzdev420) [reply src/mixins/name-request-mixin.ts:39]: I updated

Comment (severinbeauvais) [reply src/mixins/name-request-mixin.ts:39]: Thanks for the updates. Looks good!

File: src/enums/nameRequestStates.ts
Comment (severinbeauvais) [inline src/enums/nameRequestStates.ts:16]: I believe lines 26 and 27 are now obsolete. Please delete them if they're not used anywhere else. (You may have to update unit tests.)

**Review by severinbeauvais — APPROVED** [review]
Please rebase, resolve my comment, try to get another approval, and then you can merge. Thanks!

File: src/enums/nameRequestStates.ts
Comment (kzdev420) [reply src/enums/nameRequestStates.ts:16]: I updated

**Review by severinbeauvais — COMMENTED** [review]
Your app version update was cancelled by the merge. Please update it again :)

File: src/enums/nameRequestStates.ts
Comment (severinbeauvais) [reply src/enums/nameRequestStates.ts:16]: Are there no more references to the old enums, even in comments?

**Comment by kzdev420** [comment]
> Your app version update was cancelled by the merge. Please update it again :)

Do you mean to 4.7.7?

**Comment by severinbeauvais** [comment]
> > Your app version update was cancelled by the merge. Please update it again :)
> 
> Do you mean to 4.7.7?

Yes, because in main branch it's [currently 4.7.6](https://github.com/bcgov/business-edit-ui/blob/main/package.json).

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-540-bi05xbsj.web.app

**Comment by kzdev420** [comment]
> > > Your app version update was cancelled by the merge. Please update it again :)
> > 
> > 
> > Do you mean to 4.7.7?
> 
> Yes, because in main branch it's [currently 4.7.6](https://github.com/bcgov/business-edit-ui/blob/main/package.json).

I updated

**Comment by severinbeauvais** [comment]
Looks really good. One more review and then please merge this!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-540-bi05xbsj.web.app

**Review by tshyun24 — APPROVED** [review]
LGTM!

**Review by jamespaologarcia — APPROVED** [review]
LGTM

**Pull request merged**
